Broker/Publisher.php

```diff
@@ -26,7 +27,7 @@ class Publisher
     public function __construct(FactoryInterface $factory, EventDispatcherInterface $eventDispatcher, array $messageTypes = array(), LoggerInterface $logger = null)
     {
         $this->factory = $factory;
-        $this->eventDispatcher = $eventDispatcher;
+        $this->eventDispatcher = LegacyEventDispatcherProxy::decorate($eventDispatcher);
```

Comment:
The build fails, this should probably be
```suggestion
        $this->eventDispatcher = class_exists(LegacyEventDispatcherProxy::class) ? LegacyEventDispatcherProxy::decorate($eventDispatcher) : $eventDispatcher;
```

LegacyEventDispatcherProxy does not exist in old versions of sf